adapters/ddf/src/main/java/com/connexta/replication/adapters/ddf/DdfRestClientFactory.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.adapters.ddf;
+
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.MetacardTransformer;
+import java.io.IOException;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import org.apache.cxf.jaxrs.client.WebClient;
+import org.apache.cxf.jaxrs.ext.multipart.Attachment;
+import org.apache.cxf.jaxrs.ext.multipart.ContentDisposition;
+import org.apache.cxf.jaxrs.ext.multipart.MultipartBody;
+import org.apache.cxf.jaxrs.impl.MetadataMap;
+import org.codice.ddf.cxf.client.ClientFactoryFactory;
+import org.codice.ddf.cxf.client.SecureCxfClientFactory;
+import org.codice.ddf.endpoints.rest.RESTService;
+import org.codice.ddf.security.common.Security;
+
+/** Factory for creating a basic client capable of posting resources to the DDF REST endpoint. */
+public class DdfRestClientFactory {
+
+  private static final String CONTENT_DISPOSITION = "Content-Disposition";
+
+  private static final String DEFAULT_REST_ENDPOINT = "/services/catalog";
```

Comment:
✏️  Hardcoding the root context path. I though it was customizable on DDF

Reply:
Handled here: https://github.com/connexta/replication/pull/81